driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -141,26 +141,26 @@ public void selectServerAsync(final ServerSelector serverSelector, final Operati
             LOGGER.trace(format("Asynchronously selecting server with selector %s", serverSelector));
         }
         StartTime startTime = StartTime.now();
-        Timeout timeout = startServerSelectionTimeout(startTime);
+        Timeout timeout = operationContext.getTimeoutContext().startServerSelectionTimeout(startTime);
         ServerSelectionRequest request = new ServerSelectionRequest(
                 serverSelector, getCompositeServerSelector(serverSelector), timeout, startTime, callback);
 
         CountDownLatch currentPhase = phase.get();
         ClusterDescription currentDescription = description;
 
-        if (!handleServerSelectionRequest(request, currentPhase, currentDescription)) {
-            notifyWaitQueueHandler(request);
+        if (!handleServerSelectionRequest(request, currentPhase, currentDescription, operationContext)) {
+            notifyWaitQueueHandler(request, operationContext);
         }
     }
 
     @Override
-    public ClusterDescription getDescription() {
+    public ClusterDescription getDescription(final OperationContext operationContext) {
```

Comment:
Needed in:

```
    public ServerTuple selectServer(final ServerSelector serverSelector, final OperationContext operationContext) {
        isTrue("open", !isClosed());
        waitForSrv(operationContext);

Reply:
Needed for `waitForSrv`:

```
    public ServerTuple selectServer(final ServerSelector serverSelector, final OperationContext operationContext) {
        isTrue("open", !isClosed());
        waitForSrv(operationContext);
```
Since it now needs operationContext, so does getDescription:

```
    public ClusterDescription getDescription(final OperationContext operationContext) {
        isTrue("open", !isClosed());
        waitForSrv(operationContext);
```

Reply:
Ack